v4/export/dump.go

```diff
@@ -117,6 +118,11 @@ func (d *Dumper) Dump() (err error) {
 		return err
 	}
 	defer connectPool.Close()
+	writerConnPool, err := newConnectionsPool(ctx, conf.Threads, pool)
```

Comment:
I don't think we need two connection pools to parallelize splitting chunks and writing table data. It will create too many MySQL connections. In my test split chunks and dump these databases' and tables' meta SQLs usually cost less than 1s (around 300ms). The time cost to dump each tables' data is much larger than this.

If we need to parallelize this, I think an extra connection is enough.